### PR TITLE
create npmrc by setup-node instead of changesets/action

### PR DESCRIPTION
**Pull request by yukukotani**
This will be the final retry!! :bow:

changesets/action seems to create `.npmrc` only when it publish releases. So instead, create `.npmrc` by setup-node.

### CODE REVIEW

File: .github/actions/setup-pnpm/action.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ runs:
       with:
         node-version: ${{ inputs.node_version }}
         cache: pnpm
+        registry-url: "https://registry.npmjs.org"
```
Comment (yukukotani) [inline .github/actions/setup-pnpm/action.yml:20]: By specifying `registry-url`, setup-node creates `.npmrc` that refers NODE_AUTH_TOKEN. [(doc)](https://github.com/actions/setup-node#:~:text=Optional%20registry%20to%20set%20up%20for%20auth)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,6 +29,7 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (yukukotani) [inline .github/workflows/release.yml:32]: chanesets/action uses existing `.npmrc` if it does. [(doc)](https://github.com/changesets/action#:~:text=However%2C%20if%20a%20.npmrc%20file%20is%20found%2C%20the%20GitHub%20Action%20does%20not%20recreate%20the%20file)

So using `NODE_AUTH_TOKEN` to follow setup-node here.

**Review by poteboy — APPROVED** [review]
Cool

**Comment by yukukotani** [comment]
The job passed :tada: Thank you for the quick merge!

Job: https://github.com/kuma-ui/kuma-ui/actions/runs/5966781755/job/16187132773
Package: https://www.npmjs.com/package/@kuma-ui/core/v/0.0.0-dev-20230824171415

**Pull request merged**